src/io/I2C.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Contains generic implmentations for some of the I2C funcitonality.
+ * Some hardware has the ability to implement the functionality in this
+ * file so specific implementations of the I2C driver may override these
+ * methods for a hardware speedup.
+ */
+#include <EVT/io/I2C.hpp>
+
+namespace EVT::core::IO {
+
+    I2C::I2C(Pin sclPin, Pin sdaPin): sclPin(sclPin), sdaPin(sdaPin) {}
```

Comment:
Not sure if this was present in previous PRs but function definitions inside a namespace shouldn't be indented.

Reply:
Woops, yeah that should not be indented. I'll change that